src/eredis_parser.erl

```diff
@@ -174,6 +176,15 @@ do_parse({multibulk_continue, _RemainingItems, _Acc} = State, <<>>) ->
 do_parse({multibulk_continue, _RemainingItems, _Acc} = State, Data) ->
     {nested, State, Data}.
 
+%% Optimize the first clause for the most common case - when Acc is empty
+split_by_newline(Acc, Data) when size(Acc) == 0 ->
```

Comment:
Are you saying that `when size(Acc) == 0` is faster than a direct match on the empty binary?

```suggestion
split_by_newline(<<>>, Data) ->
```

Reply:
OK. It'd be good to add a code comment about this to prevent others from simplifying this in the future.